### ISSUE TITLE
Unable to export model to ONNX with dynamo and dynamic batch size

### ISSUE THREAD

**Issue by Fredrik00** [pytorch/pytorch#148886]
### 🐛 Describe the bug

I have been trying to export PARSeq, a Transformer based scene text recognition model, to ONNX with torch.onnx.export and dynamo enabled. I have been successful in getting the model exported with a fixed batch size, but unfortunately not with dynamic shapes.

I have created an input tensor matching my max batch size by repeating my original one. After passing it through the model for export, the batch size becomes fixed to the example tensor batch size rather than being dynamic. I have also attempted with dynamic_axes, which from the code looked like it would attempt to convert it to dynamic_shapes internally. 
```
image = get_dummy_input()
image_batch = image.repeat(128, 1, 1, 1)

onnx_model = torch.onnx.export(
    lightning_model,
    image_batch,
    input_names=['input'],
    output_names=['output'],
    dynamo=True,
    dynamic_shapes=[{0: torch.export.Dim('batch_size', min=1, max=128)}],
    optimize=True,
    verbose=True  # Includes metadata used during quantization
)

onnx_model.save(output_path)
```

When using the exported onnx model for inference, with a batch size of 8 in this example, I get the following error:
`onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Where node. Name:'node_Where_3144' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:540 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 8 by 128`

I couldn't find any example of the dynamic_shapes input, and the documentation is very vague. Is there something wrong with how I am specifying the dynamic input shapes?
Full example can be found here (requirements listed in requirements/onnx.txt):
https://github.com/Fredrik00/parseq/blob/tflite-export/tools/export_onnx.py

### Versions

```
[pip3] ai-edge-torch-nightly==0.4.0.dev20250303
[pip3] numpy==2.1.3
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] onnx==1.17.0
[pip3] onnxruntime==1.21.0
[pip3] onnxscript==0.2.1
[pip3] optree==0.14.1
[pip3] pytorch-lightning==2.5.0.post0
[pip3] torch==2.6.0+cu126
[pip3] torch_xla2==0.0.1.dev202412041639
[pip3] torchaudio==2.6.0+cu126
[pip3] torchmetrics==1.6.1
[pip3] torchvision==0.21.0+cu126
[pip3] triton==3.2.0
[conda] ai-edge-torch-nightly     0.4.0.dev20250303          pypi_0    pypi
[conda] numpy                     2.1.3                    pypi_0    pypi
[conda] nvidia-cublas-cu12        12.6.4.1                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.6.80                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.6.77                  pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.6.77                  pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.5.1.17                 pypi_0    pypi
[conda] nvidia-cufft-cu12         11.3.0.4                 pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.7.77                pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.7.1.2                 pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.5.4.2                 pypi_0    pypi
[conda] nvidia-cusparselt-cu12    0.6.3                    pypi_0    pypi
[conda] nvidia-nccl-cu12          2.21.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.6.85                  pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.6.77                  pypi_0    pypi
[conda] optree                    0.14.1                   pypi_0    pypi
[conda] pytorch-lightning         2.5.0.post0              pypi_0    pypi
[conda] torch                     2.6.0+cu126              pypi_0    pypi
[conda] torch-xla2                0.0.1.dev202412041639          pypi_0    pypi
[conda] torchaudio                2.6.0+cu126              pypi_0    pypi
[conda] torchmetrics              1.6.1                    pypi_0    pypi
[conda] torchvision               0.21.0+cu126             pypi_0    pypi
[conda] triton                    3.2.0                    pypi_0    pypi

```

**Comment by justinchuby** [comment]
Could be a conversion error. Could you share the onnx model? No weights needed.

**Comment by justinchuby** [comment]
Ah- there may be one bug in torch 2.6 that removes dynamic shapes during onnx export. Could you test with torch nightly?

**Comment by Fredrik00** [comment]
@justinchuby torch nightly did indeed fix the issue, thanks for the help!

I would still like to request an improvement to the documentation though. As it currently stands it was difficult to figure out the correct format for the dynamic_shapes parameter, even though I did eventually end up getting it right. A simple example included in the documentation would have been very helpful for determining if this was a bug earlier on.

**Comment by spkgyk** [comment]
Hi! I encountered the same issue. Using both 2.6 and nightly did not work.
